Astro Cloud IDE documentation (#1371)
astronomer/docs: astro/cloud-ide/overview.md

---
sidebar_label: Overview
title: Astro Cloud IDE
id: overview
description: Learn how to build pipelines with the Cloud IDE.
slug: /cloud-ide
---

import LinkCardGrid from '@site/src/components/LinkCardGrid';
import LinkCard from '@site/src/components/LinkCard';

<p class="DocItem__header-description">

>   A cloud-based, notebook-inspired IDE for writing and testing data pipelines. No Airflow knowledge or local setup required.

Reviewer: ```suggestion
  A cloud-based, notebook-inspired IDE for writing and testing data pipelines. No Airflow knowledge or local setup is required.
```

</p>

:::caution

<!-- id to make it easier to remove: cloud-ide-preview-banner -->

> The Cloud IDE is currently in _Public Preview_. This means that it is available to all Astro customers, but is still undergoing heavy development and features are subject to change. Creating projects in the Cloud IDE does not impact running tasks on existing Deployments. 

Reviewer: ```suggestion
The Cloud IDE is currently in _Public Preview_ and it is available to all Astro customers. It is still in development and features and functionality are subject to change. Creating projects in the Cloud IDE does not affect running tasks on existing Deployments. If you have feedback on the Cloud IDE you'd like to share, you can submit it on the [Astro Cloud IDE product portal](https://portal.productboard.com/75k8qmuqjacnrrnef446fggj).
```

> If you have any feedback, please submit it to the [Astro Cloud IDE product portal](https://portal.productboard.com/75k8qmuqjacnrrnef446fggj).

Reviewer: ```suggestion
If you have any feedback, please submit it to the [Astro Cloud IDE product portal](https://portal.productboard.com/75k8qmuqjacnrrnef446fggj).
```
Remove

:::

> The Astro Cloud IDE is a notebook-inspired development environment for writing and testing data pipelines with Astro. The Cloud IDE materially lowers the barrier to entry for new Airflow users and improves the development experience of experienced users.

Reviewer: ```suggestion
The Astro Cloud IDE is a notebook-inspired development environment for writing and testing data pipelines with Astro. The Cloud IDE makes it easier for new Airflow users to get started and provides experienced users with a robust development environment.
```

> One of the biggest barriers to using Airflow is writing boilerplate code for basic actions such as creating dependencies, passing data between tasks, and connecting to external services. You can configure all of these through the Astro Cloud IDE's user interface (UI) so that you only need to write the Python or SQL that executes your work.

Reviewer: ```suggestion
One of the biggest barriers to using Airflow is writing boilerplate code for basic actions such as creating dependencies, passing data between tasks, and connecting to external services. You can configure all of these with the Cloud IDE user interface (UI) so that you only need to write the Python or SQL code that executes your work.
```

> In the following screenshot, you can see how you can use the Cloud IDE to write a DAG by only writing SQL. The Astro CLoud IDE automatically generates a DAG with dependencies based only on the Jinja templating in each SQL query. All connections, package dependencies, and DAG metadata are configured through the UI.

Reviewer: ```suggestion
In the following image, you can see how you can use the Cloud IDE to write a DAG by only writing SQL. The Cloud IDE automatically generates a DAG with dependencies based only on the Jinja templating in each SQL query. All connections, package dependencies, and DAG metadata are configured with the UI.
```

![Product Screenshot](/img/cloud-ide/pipeline-editor.png)

<!-- insert demo video -->

## Cloud IDE Features

<LinkCardGrid>
  <LinkCard
    label="Focus on task logic"
    description="Turn everyday Python and SQL into Airflow-ready DAG files that follow the latest best practices."
  />
  <LinkCard
    label="Handle data seamlessly"
    description="Pass data directly from one task to another using a notebook-style interface. No configuration required."
  />
  <LinkCard
    label="Move between SQL and Python"
    description="Use SQL tables as dataframes by referencing your upstream query name, and vice versa."
  />
  <LinkCard
    label="Auto-generate your DAG"
    description="Your dependency graph and DAG file are auto-generated based on data references in your SQL and Python code."
  />
  <LinkCard
    label="Source control your changes"
    description="Push your pipeline to a Git repository with a built-in Git integration."
  />
  <LinkCard
    label="Deploy directly to Astro"

>     description="Using our out-of-the-box CI/CD, deploy your code to a production Deployment on Astro in a few clicks."

Reviewer: ```suggestion
    description="Use the included CI/CD script to deploy your code to a production Deployment on Astro."
```

  />
</LinkCardGrid>

## Documentation

<LinkCardGrid>

>   <LinkCard truncate label="Quickstart" description="Create data pipelines from scratch using the Astro Cloud IDE." href="/astro/cloud-ide/quickstart" />

Reviewer: ```suggestion
  <LinkCard truncate label="Quickstart" description="Create data pipelines with the Cloud IDE." href="/astro/cloud-ide/quickstart" />
```
